### DISCUSSION TITLE
Error when compiling LIS

### DISCUSSION THREAD

**Discussion by tais958** [NASA-LIS/LISF#1052] [Q&A] [answered]
Hi,

I was trying to compile LIS with the required libraries set up and configured. 
It was paused quickly when I type **.compile** and the error message is given below:

----------------------------------------------------------
Compiling LIS

[INFO] Compiling LIS source code

Makefile:26: Filepath: No such file or directory
Generating Filepath
rm -f *.o *.mod LIS *.d Filepath
./plugins.py
Traceback (most recent call last):
  File "./plugins.py", line 16, in <module>
    if sys.version_info.major == 2:
AttributeError: 'tuple' object has no attribute 'major'
make: *** [Filepath] Error 1

[ERR] Compile failed

-----------------------------------------------------------
Wondering if anyone knows the possible cause for this and even better if has potential solution.

Thank you,
Sheng-Lun

**Comment by tais958** [comment, marked as answer]
This issue is fixed after loaded in the python module. Thanks.

**Comment by tais958** [comment]
Sorry to activate this thread again, but it turned out another error message popped up as below:
Could anyone provide some guidance on this?

Thanks!

--------------------------------------------------------------
ifort -c  -O2 -fp-model precise -u -traceback -fpe0 -nomixed_str_len_arg -names lowercase -convert big_endian -assume byterecl -DSPMD -DHIDE_SHR_MSG -DNO_SHR_VMATH -DIFC -DLINUX -I/qfs/projects/ops/rh6/esmf/7.1/mod/modO/Linux.intel.64.intelmpi.default -DUSE_INCLUDE_MPI -I/pic/projects/sooty2/tais958/lib/eccodes-2.12.5/include/ -I/share/apps/netcdf/4.4.1.1/intel/15.0.1/include/ -I/people/scicons/constance/apps/hdf5-1.8.13/intel/15.0.1/include/ -DLIS_JULES -I./ -I../core/ -I../params/gfrac/SPORTDaily/ -I../params/gfrac/VIIRSDaily/ -I../surfacemodels/land/vic.4.1.1/ -I../surfacemodels/land/vic.4.1.1/physics/ -I../surfacemodels/land/vic.4.1.2.l/ -I../surfacemodels/land/vic.4.1.2.l/physics/ -I../surfacemodels/land/awral.6.0.0/physics/ -I../surfacemodels/land/rdhm.3.5.6/ -I../surfacemodels/land/rdhm.3.5.6/sac/ -I../surfacemodels/land/rdhm.3.5.6/frz/ -I../surfacemodels/land/rdhm.3.5.6/snow17/ -I../surfacemodels/land/rdhm.3.5.6/sachtet/ ../interp/easeV2_utils.F90
ifort -c  -O2 -fp-model precise -u -traceback -fpe0 -nomixed_str_len_arg -names lowercase -convert big_endian -assume byterecl -DSPMD -DHIDE_SHR_MSG -DNO_SHR_VMATH -DIFC -DLINUX -I/qfs/projects/ops/rh6/esmf/7.1/mod/modO/Linux.intel.64.intelmpi.default -DUSE_INCLUDE_MPI -I/pic/projects/sooty2/tais958/lib/eccodes-2.12.5/include/ -I/share/apps/netcdf/4.4.1.1/intel/15.0.1/include/ -I/people/scicons/constance/apps/hdf5-1.8.13/intel/15.0.1/include/ -DLIS_JULES -I./ -I../core/ -I../params/gfrac/SPORTDaily/ -I../params/gfrac/VIIRSDaily/ -I../surfacemodels/land/vic.4.1.1/ -I../surfacemodels/land/vic.4.1.1/physics/ -I../surfacemodels/land/vic.4.1.2.l/ -I../surfacemodels/land/vic.4.1.2.l/physics/ -I../surfacemodels/land/awral.6.0.0/physics/ -I../surfacemodels/land/rdhm.3.5.6/ -I../surfacemodels/land/rdhm.3.5.6/sac/ -I../surfacemodels/land/rdhm.3.5.6/frz/ -I../surfacemodels/land/rdhm.3.5.6/snow17/ -I../surfacemodels/land/rdhm.3.5.6/sachtet/ ../interp/map_utils.F90
ifort -c  -O2 -fp-model precise -u -traceback -fpe0 -nomixed_str_len_arg -names lowercase -convert big_endian -assume byterecl -DSPMD -DHIDE_SHR_MSG -DNO_SHR_VMATH -DIFC -DLINUX -I/qfs/projects/ops/rh6/esmf/7.1/mod/modO/Linux.intel.64.intelmpi.default -DUSE_INCLUDE_MPI -I/pic/projects/sooty2/tais958/lib/eccodes-2.12.5/include/ -I/share/apps/netcdf/4.4.1.1/intel/15.0.1/include/ -I/people/scicons/constance/apps/hdf5-1.8.13/intel/15.0.1/include/ -DLIS_JULES -I./ -I../core/ -I../params/gfrac/SPORTDaily/ -I../params/gfrac/VIIRSDaily/ -I../surfacemodels/land/vic.4.1.1/ -I../surfacemodels/land/vic.4.1.1/physics/ -I../surfacemodels/land/vic.4.1.2.l/ -I../surfacemodels/land/vic.4.1.2.l/physics/ -I../surfacemodels/land/awral.6.0.0/physics/ -I../surfacemodels/land/rdhm.3.5.6/ -I../surfacemodels/land/rdhm.3.5.6/sac/ -I../surfacemodels/land/rdhm.3.5.6/frz/ -I../surfacemodels/land/rdhm.3.5.6/snow17/ -I../surfacemodels/land/rdhm.3.5.6/sachtet/ ../core/LIS_PRIV_tileMod.F90
ifort -c  -O2 -fp-model precise -u -traceback -fpe0 -nomixed_str_len_arg -names lowercase -convert big_endian -assume byterecl -DSPMD -DHIDE_SHR_MSG -DNO_SHR_VMATH -DIFC -DLINUX -I/qfs/projects/ops/rh6/esmf/7.1/mod/modO/Linux.intel.64.intelmpi.default -DUSE_INCLUDE_MPI -I/pic/projects/sooty2/tais958/lib/eccodes-2.12.5/include/ -I/share/apps/netcdf/4.4.1.1/intel/15.0.1/include/ -I/people/scicons/constance/apps/hdf5-1.8.13/intel/15.0.1/include/ -DLIS_JULES -I./ -I../core/ -I../params/gfrac/SPORTDaily/ -I../params/gfrac/VIIRSDaily/ -I../surfacemodels/land/vic.4.1.1/ -I../surfacemodels/land/vic.4.1.1/physics/ -I../surfacemodels/land/vic.4.1.2.l/ -I../surfacemodels/land/vic.4.1.2.l/physics/ -I../surfacemodels/land/awral.6.0.0/physics/ -I../surfacemodels/land/rdhm.3.5.6/ -I../surfacemodels/land/rdhm.3.5.6/sac/ -I../surfacemodels/land/rdhm.3.5.6/frz/ -I../surfacemodels/land/rdhm.3.5.6/snow17/ -I../surfacemodels/land/rdhm.3.5.6/sachtet/ ../core/LIS_mpiMod.F90
../core/LIS_mpiMod.F90(22): error #5102: Cannot open include file 'mpif.h'
  include 'mpif.h'
----------^
compilation aborted for ../core/LIS_mpiMod.F90 (code 1)
make: *** [LIS_mpiMod.o] Error 1

[ERR] Compile failed

-------------------------------------------------------------------------------

**Reply by bmcandr** [reply]
Hi @tais958,

Our expert on compiling LISF is out on leave at the moment, but I will try to help. Try this: change the environment variables `$FC` and `$CC` to `mpiifort` and `mpicc`, respectively. You will also need to add the path to your Intel MPI installation's `bin` directory to your `$PATH` if it is not already included.

**Reply by tais958** [reply]
Hi @bmcandr ,

Thanks so much for the reply. I tried what you suggested and it did work! 
However, it was paused at one point that seems to be relevant to error in ESMF as below:

/qfs/projects/ops/rh6/esmf/7.1/lib/libO/Linux.intel.64.intelmpi.default/libesmf.so: undefined reference to `dgelsd_'
/qfs/projects/ops/rh6/esmf/7.1/lib/libO/Linux.intel.64.intelmpi.default/libesmf.so: undefined reference to `ilaenv_'
/qfs/projects/ops/rh6/esmf/7.1/lib/libO/Linux.intel.64.intelmpi.default/libesmf.so: undefined reference to `dgelsy_'

I wonder if you or anyone has seen this before and could provide any guidence?
The ESMF was built with the same compiler as other libraries are. Not sure if anything I may miss to check.

Thanks!
Sheng-Lun

**Reply by bmcandr** [reply]
I have not seen this error before. It looks like the errors are related to the use of LAPACK, an optional library, by ESMF. See here for more: https://earthsystemmodeling.org/docs/release/latest/ESMF_usrdoc/node10.html#SECTION000104100000000000000

Did you happen to set `$ESMF_LAPACK` when building ESMF?

**Reply by tais958** [reply]
Thanks, it was due to the LAPACK routines were not linked in from MLK. It is resolved after adding "-mkl" to the end.
